codex/node.nim

```diff
@@ -441,11 +437,13 @@ proc start*(node: CodexNodeRef) {.async.} =
         return failure(error)
 
       trace "Fetching block for manifest", cid
-      # TODO: This will probably require a call to `getBlock` either way,
-      # since fetching of blocks will have to be selective according
-      # to a combination of parameters, such as node slot position
-      # and dataset geometry
-      if fetchErr =? (await node.fetchBatched(manifest, onBatch = onBatch, expiry = some request.expiry.toSecondsSince1970)).errorOption:
+      let expiry = request.expiry.toSecondsSince1970
+      proc expiryUpdateOnBatch(blocks: seq[bt.Block]): Future[void] {.async.} =
+        let ensureExpiryFutures = blocks.mapIt(node.blockStore.ensureExpiry(it.cid, expiry))
+        await allFuturesThrowing(ensureExpiryFutures)
```

Comment:
This will throw on a failed future and since this is a callback it will most likely crash the process, we should surround this with a `try/except` or use `catch` from results to convert it into a result.

Reply:
Good point! Thanks for the catch! 👍